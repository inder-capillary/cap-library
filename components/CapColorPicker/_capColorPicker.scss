@import "../styles/_variables.scss";

$cap-slider: cap-slider-v2;

.#{$cap-slider} {
  .react-colorful {
    width: 100%;
    height: 100%;
  }
  .react-colorful__saturation {
    display: none;
  }
  .react-colorful__last-control {
    border-radius: 2px;
  }
}