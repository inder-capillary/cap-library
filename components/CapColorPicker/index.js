/**
*
* CapColorPicker - component used to select color
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import './_capColorPicker.scss';
import { HexColorPicker } from "react-colorful";

const classNames = require('classnames');

const clsPrefix = "cap-colorpicker-v2";

class CapColorPicker extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {className, color, setColor, hexSelector } = this.props;
    const colorPickerType = hexSelector ? "cap-hexslider-v2" : "cap-slider-v2";

    return (
      <div className={classNames(clsPrefix, className, colorPickerType)}>
        <HexColorPicker color={color} onChange={setColor} />
      </div>
    );
  }
}

CapColorPicker.defaultProps = {
  className: '',
  color: '#d42020',
  setColor: () => {},
  hexSelector: false,
};

CapColorPicker.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  setColor: PropTypes.func,
  hexSelector: PropTypes.bool,
};

export default CapColorPicker;
