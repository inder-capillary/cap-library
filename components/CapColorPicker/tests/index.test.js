import React from 'react';
import {shallow} from 'enzyme';
import CapColorPicker from '../index';

describe('<CapColorPicker />', () => {
  it('CapColorPicker renders correctly', () => {
    const tree = shallow(<CapColorPicker />);
    expect(tree).toMatchSnapshot();
  });

  it('render size input correctly with small value', () => {
    const fn = () => {};
    const props = {
      className: 'test',
      color: '#d42024',
      setColor: fn,
      hexSelector: false,
    };
    const CapColorPickerComponent = shallow(<CapColorPicker {...props} />);
    expect(CapColorPickerComponent.instance().props.className).toEqual("test");
    expect(CapColorPickerComponent.instance().props.color).toEqual('#d42024');
    expect(CapColorPickerComponent.instance().props.setColor).toEqual(fn);
    expect(CapColorPickerComponent.instance().props.hexSelector).toEqual(false);
  });
});
