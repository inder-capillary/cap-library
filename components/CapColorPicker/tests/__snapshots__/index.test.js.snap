// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapColorPicker /> CapColorPicker renders correctly 1`] = `
<div
  className="cap-colorpicker-v2 cap-slider-v2"
>
  <Component
    color="#d42020"
    onChange={[Function]}
  />
</div>
`;
